packages/lisk-p2p/src/peer.ts

```diff
@@ -12,59 +12,167 @@
  * Removal or modification of this copyright notice is prohibited.
  *
  */
-/* tslint:disable:interface-name variable-name */
-import { PeerState } from './p2p_types';
+/* tslint:disable:interface-name */
+import {
+	P2PRequestPacket,
+	P2PMessagePacket,
+	P2PResponsePacket,
+	NodeInfo,
+} from './p2p_types';
 
-// TODO: Use to create outbound socket connection inside peer object.
-// TODO: const socketClusterClient = require('socketcluster-client');
+import socketClusterClient from 'socketcluster-client';
 
 export interface PeerConfig {
 	readonly clock?: Date;
-	readonly height: number;
+	readonly height?: number;
 	readonly id: string;
 	/* tslint:disable:next-line: no-any */
 	readonly inboundSocket?: any; // TODO: Type SCServerSocket
 	readonly ipAddress: string;
 	readonly os?: string;
 	readonly version?: string;
 	readonly wsPort: number;
+	readonly nodeInfo: NodeInfo;
+}
+
+export enum ConnectionState {
+	CONNECTING = 0,
+	CONNECTED = 1,
+	DISCONNECTED = 2,
+}
+
+export interface PeerState {
```

Comment:
there are so many names revolving around Peer, maybe we need a descriptive name, maybe `PeerConnnectionState`?